### DISCUSSION TITLE
Trying to understand -P windows-latest=-self-hosted

### DISCUSSION THREAD

**Discussion by frederick-brier** [nektos/act#1984] [Q&A] [answered]
My goal is to be able to navigate inside the windows-latest/windows-2022 Github Actions image. I used Act to get my workflow running on my local workstation (Windows 10). After pushing the changes to the repo, there is a failure finding VS2022 files. I tried the --reuse flag to keep the docker image running. "docker ps" doesn't show any images running. My questions therefore are:

- When you specify windows-latest=-self-hosted, is it actually a docker image used by Github Actions, or not an image at all? Is it just running on your workstation Windows instance, but with checked-out code in C:\Users\<user-dir>\.cache\act ?
- Does Act even support running in a Windows docker image?
- Is there a way to have "docker exec" specify the Github docker image windows-2022 or windows-latest? "docker exec -it windows-latest powershell.exe fails. I have also tried a number of variants of "docker pull ghcr.io/github/windows-2022".

Thank you

**Comment by ChristopherHX** [comment, marked as answer]
`-self-hosted` has nothing to do with docker, it is for platforms without a docker engine like freebsd, plan9, dragonflybsd, solaris, aix....

> Is it just running on your workstation Windows instance, but with checked-out code in C:\Users<user-dir>.cache\act ?

yes, exactly

> Does Act even support running in a Windows docker image?

No, but a tool similar to act which supports this exists


> Is there a way to have "docker exec" specify the Github docker image windows-2022 or windows-latest?

GitHub does not provide such docker images, they use VM's and don't upload the osdisk (you have to build from source in azure actions/runner-images )

**Reply by wmmc88** [reply]
> No, but a tool similar to act which supports this exists
Which tool are you referring to?

**Reply by ChristopherHX** [reply]
>> No, but a tool similar to act which supports this exists
>
> Which tool are you referring to?

I made a tool called `runner.server` under my GitHub profile, where I patched actions/runner to execute windows container images (requires docker desktop beeing in windows container mode)

You can find it with google/bing etc. (no links, due to conflict of interest with this project)
